File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k=0
        i=0
        while(i<len(nums)):
            if nums[i]==0:
                nums.pop(i)
                k+=1
                continue
            i+=1
            
        nums.extend([0]*k)

        

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& arr) {
        int i = -1;
        for (int j = 0; j < arr.size(); j++) {
            if (arr[j] == 0) {
                i = j;
                break;
            }
        }
        if (i == -1)
            return;
        for (int j = i+1; j < arr.size(); j++) {
            if (arr[j] != 0) {
                swap(arr[i], arr[j]);
                 i++;
            }
        }
    }
};